### PR TITLE
impr(exchnage): add support for old version for exchange app

### PR DESCRIPTION
**Pull request by mattwasielewski**

### CODE REVIEW

File: scripts/deploy.py [hunk 1]
```diff
@@ -91,6 +95,13 @@ def get_contract_txn(contract_type: str, global_schema, **format_data):
     prompt="Fee bps",
     help="Fee in basis points taken from the outcome of each swap.",
 )
+@click.option(
+    "--version",
+    type=int,
+    default=1,
```
Comment (tulustul) [inline scripts/deploy.py]: Maybe `default=None`, which will pick the current version (the one without `_v_x` suffix).
That way, we won't have to update this place with each new version.

File: scripts/deploy.py [hunk 2]
```diff
@@ -117,20 +128,23 @@ def deploy_contract(
...
     pact_fee_bps: int,
     amplifier: int,
     admin_and_treasury_address: str,
 ):
     print("EC deployment begins")
-
+    teal_version = 6 if version else 5
     if contract_type == 'constant_product':
-        global_schema = StateSchema(9, 4)
+        global_schema = StateSchema(9, 4) if version else StateSchema(4, 1)
```
Comment (tulustul) [inline scripts/deploy.py]: Let's make a dict.
```
{
    ('constant_product', 1): StateSchema(...),
    ...
}
```

**Review by tulustul — APPROVED** [review]

File: scripts/deploy.py (hunk 1 above)
Comment (mattwasielewski) [reply scripts/deploy.py]: DONE.

File: scripts/deploy.py (hunk 2 above)
Comment (mattwasielewski) [reply scripts/deploy.py]: DONE

**Pull request merged**